src/compiler/tsbuild.ts

```diff
@@ -294,8 +294,10 @@ namespace ts {
         return changeExtension(outputPath, (fileExtensionIs(inputFileName, Extension.Tsx) && configFile.options.jsx === JsxEmit.Preserve) ? Extension.Jsx : Extension.Js);
     }
 
+    const extensionsWithNoOutput: ReadonlyArray<string> = [Extension.Dts, Extension.Json];
     function getOutputFileNames(inputFileName: string, configFile: ParsedCommandLine): ReadonlyArray<string> {
-        if (configFile.options.outFile) {
+        // outFile is handled elsewhere; .d.ts files don't generate outputs

```

Comment:
Json files are emitted just like js files. But they do not have sourcemaps.

Reply:
Really? So if I `require`/`import` my package.json it'll bet included in the emit? o.O

Reply:
Count me as surprised as well, but verified and this is the case